### PR TITLE
HOCS-5606: Ability to search by Application Reference

### PR DESCRIPTION
**Pull request by AlexW013**

### CODE REVIEW

**Comment by robertdeniszczyc2** [comment]
Hi @AlexW013 ,

Rather than applying the query type on the field here, I wonder if it might be better to set the query type on the field where the field is defined in hocs-frontend, which would then allow us to set different query types on fields with the same name depending on case type. In this current implementation I think the query type configuration would be applied to every user with that field, regardless of case type.

Happy to talk through,

Cheers,
Rob

File: src/main/java/uk/gov/digital/ho/hocs/search/api/CaseDataService.java [hunk 1]
```diff
@@ -56,12 +57,16 @@ public class CaseDataService {
 
     private final CaseTypeMappingRepository caseTypeMappingRepository;
 
+    private final FieldQueryTypeMappingRepository fieldQueryTypeMappingRepository;
```
Comment (SWhitlam) [inline src/main/java/uk/gov/digital/ho/hocs/search/api/CaseDataService.java:60]: I wonder if we should inject this into the HocsQueryBuilder class? It doesn't appear to be used within the case data service. Or alternatively get the query type from the repository within this method and pass the search type through to the query builder?

**Comment by AlexW013** [comment]
> Hi @AlexW013 ,
> 
> Rather than applying the query type on the field here, I wonder if it might be better to set the query type on the field where the field is defined in hocs-frontend, which would then allow us to set different query types on fields with the same name depending on case type. In this current implementation I think the query type configuration would be applied to every user with that field, regardless of case type.
> 
> Happy to talk through,
> 
> Cheers, Rob

Hi @robertdeniszczyc2,
I see your point about case type. I did have a call with Liam and Stuart around this ticket before I went with the current implementation but happy to discuss changing on the frontend if you think this needs rethinking

Cheers Alex

**Comment by robertdeniszczyc2** [comment]
> 

Ah no problem, if we've already had the discussion happy to proceed with this 👍

**Comment by LiamMacP** [comment]
> Hi @AlexW013 ,
> 
> Rather than applying the query type on the field here, I wonder if it might be better to set the query type on the field where the field is defined in hocs-frontend, which would then allow us to set different query types on fields with the same name depending on case type. In this current implementation I think the query type configuration would be applied to every user with that field, regardless of case type.
> 
> Happy to talk through,
> 
> Cheers, Rob

As Alex has pointed out we discussed this but decided on this route for now with room to redesign going forward. 

There would be other issues with having it in the frontend - like the reducing of fields down for the UI. Do you send a wildcard one or a non wildcard if 2 case types have the same field. I'm also unsure if I believe its a frontend concern (at the moment - pending a total search refactor). 

If this starts becoming more of an issue we would likely just implement case type based fields so one can have wildcard and others would use match. That also invites further complications. 

Maybe its something to look at though when we get time to look at rewriting the search UI.

**Comment by robertdeniszczyc2** [comment]
> > Hi @AlexW013 ,
> > Rather than applying the query type on the field here, I wonder if it might be better to set the query type on the field where the field is defined in hocs-frontend, which would then allow us to set different query types on fields with the same name depending on case type. In this current implementation I think the query type configuration would be applied to every user with that field, regardless of case type.
> > Happy to talk through,
> > Cheers, Rob
> 
> As Alex has pointed out we discussed this but decided on this route for now with room to redesign going forward.
> 
> There would be other issues with having it in the frontend - like the reducing of fields down for the UI. Do you send a wildcard one or a non wildcard if 2 case types have the same field. I'm also unsure if I believe its a frontend concern (at the moment - pending a total search refactor).
> 
> If this starts becoming more of an issue we would likely just implement case type based fields so one can have wildcard and others would use match. That also invites further complications.
> 
> Maybe its something to look at though when we get time to look at rewriting the search UI.

Since between us we've all separately had the same discussion it's probably worth revisiting, but agree that can come later and not in this scope.

**Review by LiamMacP — APPROVED** [review]
It's okay as a first pass - maybe we could leverage Spring beans for the query builder in the future and have that own this.

**Comment by AlexW013** [comment]
> It's okay as a first pass - maybe we could leverage Spring beans for the query builder in the future and have that own this.

Thanks Liam. I am not authorised to merge this PR for some reason

**Pull request merged**